zeus/monitor/energy.py

```diff
@@ -208,18 +227,34 @@ def begin_window(self, key: str, sync_cuda: bool = True) -> None:
 
         # Freeze the start time of the profiling window.
         timestamp: float = time()
-        energy_state: dict[int, float] = {}
+        gpu_energy_state: dict[int, float] = {}
         for gpu_index in self.gpu_indices:
             # Query energy directly if the GPU has newer architecture.
             # Otherwise, the Zeus power monitor is running in the background to
             # collect power consumption, so we just need to read the log file later.
             if self.gpus.supportsGetTotalEnergyConsumption(gpu_index):
-                energy_state[gpu_index] = (
+                gpu_energy_state[gpu_index] = (
                     self.gpus.getTotalEnergyConsumption(gpu_index) / 1000.0
                 )
 
+        cpu_energy_state: dict[int, float] = {}
+        dram_energy_state: dict[int, float] = {}
+        for cpu_index in self.cpu_indices:
+            cpu_measurement = self.cpus.getTotalEnergyConsumption(cpu_index) / 1000.0
+            cpu_energy_state[cpu_index] = cpu_measurement.cpu_mj
+            if (
+                self.cpus.supportsGetSubpackageEnergyConsumption(cpu_index)
+                and cpu_measurement.dram_mj is not None
+            ):
+                dram_energy_state[cpu_index] = cpu_measurement.dram_mj
```

Comment:
This is inefficient, because `self.cpus.supportsGetSubpackageEnergyConsumption`'s return value is always the same. If calling this is necessary, it should be called once inside `__init__` and cached.

However, do you even have to call this at all? Isn't it sufficient to check whether `cpu_measurement.dram_mj is None`?

Reply:
Looks like the same question applies to `end_window`.

Reply:
Changed the logic so that we only check `if cpu_measurement.dram_mj is not None` in `start_window` and `if dram_start_energy is not None and cpu_measurement.dram_mj is not None` in `end_window`

Reply:
I can see that as I did another review. Just resolve comments if they were handled -- all three that you just commented on. I'm referring to other comments. There's like 9 left that you haven't even responded or addressed.